ansible/roles/gpu/tasks/nvidia-gpu.yaml

```diff
@@ -48,6 +48,58 @@
     path: "{{ nvidia_remote_bundle_path }}"
     state: directory
 
+- name: find NVIDIA driver installer file
+  find:
+    paths: "{{ nvidia_remote_bundle_path }}"
+    patterns: "NVIDIA-Linux-x86_64-*.run"
+  register: nvidia_driver_files
+
+- name: extract version from file names
+  set_fact:
+    nvidia_driver_versions: "{{ nvidia_driver_files.files | map(attribute='path') | map('regex_replace', '^.*NVIDIA-Linux-x86_64-(\\d+\\.\\d+\\.\\d+)\\.run$', '\\1') | list }}"
+    uninstall: false
```

Comment:
lazily sets the fact here to false so if there is no version < nvidia_driver_version the next block doesn't complain about `uninstall` not being set.